src/main/java/com/endava/internship/service/UserServiceImpl.java

```diff
@@ -4,66 +4,84 @@
 import com.endava.internship.domain.User;
 import com.endava.internship.service.UserService;
 
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 public class UserServiceImpl implements UserService {
 
     @Override
     public List<String> getFirstNamesReverseSorted(List<User> users) {
-        throw new UnsupportedOperationException("Not implemented");
+        return users.stream().map(User::getFirstName).
```

Comment:
put please all methods in new line, except stream()

Reply:
has been changed